dali/test/python/test_operator_numpy_reader.py

```diff
@@ -0,0 +1,134 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali as dali
+import numpy as np
+from numpy.testing import assert_array_equal, assert_allclose
+import os
+import tempfile
+
+class NumpyReaderPipeline(Pipeline):
+    def __init__(self, path, batch_size, path_filter="*.npy", num_threads=1, device_id=0, num_gpus=1):
+        super(NumpyReaderPipeline, self).__init__(batch_size,
+                                                  num_threads,
+                                                  device_id)
+        
+        self.input = ops.NumpyReader(file_root = path,
+                                     file_filter = path_filter,
+                                     shard_id = device_id,
+                                     num_shards = num_gpus)
+
+        
+    def define_graph(self):
+        inputs = self.input(name="Reader")
+
+        return inputs
+
+# test: compare reader with numpy
+# with different batch_size and num_threads
+
+def test_reader_vs_numpy():
+
+    # create temporary directory
+    with tempfile.TemporaryDirectory() as test_data_root:
+    
+        # we need that
+        rng = np.random.RandomState(12345)
+    
+        # 1D array integer
+        arr = rng.randint(0,10, (15))
+        np.save(os.path.join(test_data_root, "test_01.npy"), arr)
+
+        # 1D array float
+        arr = rng.random_sample((25)).astype(np.float32)
+        np.save(os.path.join(test_data_root, "test_02.npy"), arr)
+
+        # 1D array double
+        arr = rng.random_sample((8)).astype(np.float64)
+        np.save(os.path.join(test_data_root, "test_03.npy"), arr)
+
+        # 2D array float
+        arr = rng.random_sample((25,10)).astype(np.float32)
+        np.save(os.path.join(test_data_root, "test_04.npy"), arr)
+
+        # 3D array double
+        arr = rng.random_sample((1,20,10)).astype(np.float64)
+        np.save(os.path.join(test_data_root, "test_05.npy"), arr)
+
+        # 3D array float fortran order
+        arr = rng.random_sample((3,4,2)).astype(np.float32)
+        arr = np.asfortranarray(arr)
+        np.save(os.path.join(test_data_root, "test_06.npy"), arr)
+
+        # 3D Fortran with singleton dim
+        arr = rng.randint(0, 10, (3,1,2))
+        arr = np.asfortranarray(arr)
+        np.save(os.path.join(test_data_root, "test_07.npy"), arr)
+
+        # singleton file
+        arr = 3
+        np.save(os.path.join(test_data_root, "test_08.npy"), arr)
+    
+        # get files:
+        filelist = sorted([os.path.join(test_data_root, x)
+                           for x in os.listdir(test_data_root) if x.endswith('.npy')])
+
+        # batch size 1:
+        for num_threads in [1, 2, 4, 8]:
```

Comment:
Maybe you can use https://nose.readthedocs.io/en/latest/writing_tests.html#test-generators.
If it fails you would not when, not it is some big and not very verbose test.

Reply:
Fixed